portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,19 +26,41 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  
+
+  //ArrayList<String> inputs = new ArrayList<String>();
+  ArrayList<String> names = new ArrayList<String>();
+  ArrayList<String> comments = new ArrayList<String>();
+
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+  public void doGet(HttpServletRequest request, HttpServletResponse response) 
+  throws IOException {
     response.setContentType("application/json;");
-    List<String> messages = new ArrayList<String>();
-    messages.add("ElementNumOne");
-    messages.add("ElementNumTwo");
-    messages.add("ElementNumThree");
 
-    Gson gson = new Gson();
-    String json = gson.toJson(messages);
+    /**Gson gson = new Gson();
+    String output = gson.toJson(inputs);
+    response.getWriter().println(output);**/
+  
+    //patches together the name and comment
+    for (int i = 0; i < names.size(); i++) {
+      Gson gson = new Gson();
+      String userName = gson.toJson(names.get(i));
+      String userComment = gson.toJson(comments.get(i));
```

Comment:
I don't know if two array lists is the best data structure for this scenario. You'll always want to associate names.get(i) and comments.get(i), so it would be good to make that association less fragile (i.e. make them part of the same structure). There are a couple of ways to do this, but I would suggest making some sort of small object to encompass that information.

Reply:
Do you suggest making an object  (eg User)that contains, for example, a username and a comment and then I can iterate through those users? Or rather make an arrayList that contains both the username and the comment separated by a special character or something

Reply:
I think a whole object for a User might be overkill at this point (but good thought!), since we don't really have a firm concept of a 'user' yet, and that brings up a lot of questions about what a User is and if you should track names and combine comments for the same user. I was thinking an even simpler object than that, to just track a single comment.

Saving the string 'userName + ":" + userComment' in a list isn't a bad idea for this CL, however I don't know how sustainable that will be, especially since you're saving the name and comment separately in the entity in your next CL: https://github.com/adamfweidman/myStepPortfolio/pull/7/files.